api/v1alpha1/gatewayclass_types.go

```diff
@@ -61,6 +61,29 @@ type GatewayClassSpec struct {
 	// +required
 	Controller string `json:"controller" protobuf:"bytes,1,opt,name=controller"`
 
+	// GatewayNamespaceSelector is a selector of valid namespaces that Gateways
```

Comment:
This does not quite match the description in `apimachinery/pkg/apis/meta/v1/types.go`: 

Can we reference and make the two descriptions identical?

```
1073 // A label selector is a label query over a set of resources. The result of matchLabels and
1074 // matchExpressions are ANDed. An empty label selector matches all objects. A null
1075 // label selector matches no objects.
1076 type LabelSelector struct {
```

Reply:
For each of these, we also need details for controllers too.

maybe something like:
```
// Controllers must not allow Gateways in namespaces outside this selector.
// Controllers must surface the namespaces selected using <some status thing we haven't done yet>
```

Similarly for AllowedRouteNamespaces.

Reply:
Updated, thanks! Have not added a reference to status as part of this, but agree that should be added soon.